src/commands/run.cr

```diff
@@ -0,0 +1,24 @@
+require "./command"
+
+module Shards
+  module Commands
+    class Run < Command
+      def run(targets, options, run_options)
+        targets = spec.targets.map(&.name) if targets.empty?
+        raise Error.new("Error No targets defined.") if targets.empty?
+        raise Error.new("Error More than 1 target defined. Must pass target name as parameter.") if targets.size > 1

```

Comment:
We can improve the error mesages, here. For example we can know whether the user asked for a target or not. In the former case we could say something like "please specify one target only; if you meant to pass arguments you may use: shards run target -- args". In the latter case we could say "please specify which target you want to run" with maybe a list.

Reply:
Implemented.